From 2688feb2742a6021ca211ae5c106b12c3d822946 Mon Sep 17 00:00:00 2001
From: Danny Milosavljevic <dannym@scratchpost.org>
Date: Mon, 11 Feb 2019 16:07:12 +0100
Subject: [PATCH] codec: vpx: Detect libvpx 1.8.0 and, if detected, use fewer
 frame formats in the chroma_table

Signed-off-by: Steve Lhomme <robux4@ycbcr.xyz>
(cherry picked from commit 5575fe3eb3fd46bada8662268b74d03493476a84)

Downloaded from upstream commit
https://git.videolan.org/?p=vlc/vlc-3.0.git;a=commitdiff;h=2688feb2742a6021ca211ae5c106b12c3d822946

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 modules/codec/vpx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/codec/vpx.c b/modules/codec/vpx.c
index 2b1e37818d..aba180361f 100644
--- a/modules/codec/vpx.c
+++ b/modules/codec/vpx.c
@@ -116,6 +116,7 @@ static const struct
     { VLC_CODEC_I440, VPX_IMG_FMT_I440, 8, 0 },
 
     { VLC_CODEC_YV12, VPX_IMG_FMT_YV12, 8, 0 },
+#if VPX_IMAGE_ABI_VERSION < 5
     { VLC_CODEC_YUVA, VPX_IMG_FMT_444A, 8, 0 },
     { VLC_CODEC_YUYV, VPX_IMG_FMT_YUY2, 8, 0 },
     { VLC_CODEC_UYVY, VPX_IMG_FMT_UYVY, 8, 0 },
@@ -128,7 +129,7 @@ static const struct
 
     { VLC_CODEC_ARGB, VPX_IMG_FMT_ARGB, 8, 0 },
     { VLC_CODEC_BGRA, VPX_IMG_FMT_ARGB_LE, 8, 0 },
-
+#endif
     { VLC_CODEC_GBR_PLANAR, VPX_IMG_FMT_I444, 8, 1 },
     { VLC_CODEC_GBR_PLANAR_10L, VPX_IMG_FMT_I44416, 10, 1 },
 
-- 
2.11.0

